src/pymorize/units.py

```diff
@@ -9,189 +9,253 @@
+    """
+    model_unit = rule.get("model_unit", None)
+    from_unit = da.attrs.get("units", None)
+    if model_unit is not None:
+        logger.info(
+            f"user defined units {model_unit!r} takes precedence"
+            f" over units defined in dataset {from_unit!r}"
+        )
+        from_unit = model_unit
+    to_unit = rule.data_request_variable.units
+    to_unit_alias = None
+    cmor_variable = rule.data_request_variable.variable_id
+    dimless_mapping = rule.get("dimensionless_unit_mappings", {})
+    if cmor_variable in dimless_mapping:
+        try:
+            to_unit_alias = dimless_mapping.get(cmor_variable)[to_unit]
+        except KeyError:
+            raise KeyError("Dimensionless unit not found in mappings")
+        if to_unit_alias is not None:
+            logger.info(
+                f"unit alias {to_unit_alias!r} used for representing {to_unit!r}."
+                f" see dimensionless variable map for variable {cmor_variable!r}"
+            )
+    if from_unit is None:
+        raise ValueError("Unit not defined")
+    if not (to_unit or to_unit_alias):
+        raise ValueError("Unit not defined")
+    return from_unit, to_unit, to_unit_alias
+
+
 def handle_chemicals(
-    s: Union[str, None] = None, pattern: Pattern = re.compile(r"mol(?P<symbol>\w+)")
+    s: Union[str, None], pattern: Pattern = re.compile(r"mol(?P<symbol>\w+)")
 ):
-    """Registers known chemical elements definitions to global ``ureg`` (unit registry)
+    """
+    Handle units containing chemical symbols.
+
+    If the unit string contains a chemical symbol (e.g. molNaCl), Pint will
+    raise an error because it does not know the definition of the chemical
+    symbol. This function attempts to detect chemical symbols in the unit
+    string and register a unit definition for it with the aid of chemicals
+    package.
 
     Parameters
     ----------
-    s: str or None
-        string to search for chemical elements based upon the symbol, e.g. ``C`` for carbon.
-    pattern: re.Pattern
-        compiled regex pattern to search for chemical elements. This should contain a
-        `named group <https://docs.python.org/3/howto/regex.html#non-capturing-and-named-groups>`_ ``symbol``
-        to extract the symbol of the chemical element from a potentially larger string.
+    s : str
+        The unit string to parse.
+    pattern : re.Pattern, optional
+        The regular expression pattern to use for searching for chemical
+        symbols in the unit string. Defaults to a pattern that matches
+        "mol" followed by any number of word characters.
+
+    Returns
+    -------
+    None
 
     Raises
     ------
     ValueError
-        If the chemical element is not found in the periodic table.
-
-    See Also
-    --------
-    ~chemicals.elements.periodic_table: Periodic table of elements
-    ~re.compile: `Python's regex syntax <https://docs.python.org/3/library/re.html#regular-expression-syntax>`_.
```

Comment:
Why was this removed? I feel it is goo to have a cross-link to the package, for reference purposes.

Reply:
fixed